public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/completions.ts

```diff
@@ -380,14 +380,26 @@ export async function getLogfmtCompletions(
     completions = [...completions, ...parserCompletions, ...pipeOperations];
   }
 
-  const labelPrefix = otherLabels.length === 0 || trailingComma ? '' : ', ';
   const labels = extractedLabelKeys.filter((label) => !otherLabels.includes(label));
+
+  // No other labels or trailing comma, so we don't need to add a prefix
```

Comment:
It looks like this only works for the first label in the list, on this branch I'm seeing the same behavior on 2nd+ label

https://github.com/grafana/grafana/assets/109082771/912eba47-70af-4eb1-b102-38f6b9c520c5

Start with a query with multiple labels passed into logfmt
`{app_kubernetes_io_name="apisix"} | logfmt  job , name , namespace`
And then reproduce the issue by deleting a few chars from the last label, and then selecting the autocomplete.
`{app_kubernetes_io_name="apisix"} | logfmt job , name , , namespace`

Reply:
It's very odd because I tested the same and I can only reproduce the extra pipe that you show in the video, but that is unrelated (which now that we're at it I will try to improve that one as well).

As for the labels:

https://github.com/grafana/grafana/assets/1069378/3ae64a74-a750-40ec-bbff-4d62b8030b0c

Reply:
I checked out the branch again and I'm still seeing this:

https://github.com/grafana/grafana/assets/109082771/dd45694c-1a29-435d-bc41-cb2467354f90

The simplest reproduction is to start with a query with multiple labels, and then delete a few chars from the last label, and then select the autocompletion result.